### PR TITLE
chore: remove soon to be deprecated gradle configs and use recommended configs

### PR DESCRIPTION
**Pull request by Gogetter**
### JIRA link (if applicable) ###
[FPLA-1657](https://tools.hmcts.net/jira/browse/FPLA-1657)

### Change description ###
Remove soon to be deprecated gradle configs and use recommended configs

- compile -> implementation
- testCompile -> testImplementation
- performanceTestCompile -> performanceTestImplementation
- and others

### Notes ###
Not all dependencies have been updated to use `implementation`, `testImplementation`. This is because not all [dependencies/plugins](https://github.com/gradle/gradle/issues/11769#issuecomment-570173947) currently have support for `gradle-6.4.1`.

[gradlew](https://github.com/hmcts/fpl-ccd-configuration/blob/cb308dd7d0128d9d58b1d60f817bdfc80e6907e0/gradlew) and [gradlew.bat](https://github.com/hmcts/fpl-ccd-configuration/blob/cb308dd7d0128d9d58b1d60f817bdfc80e6907e0/gradlew.bat) generated using `gradle wrapper --gradle-version 6.4.1`

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java [hunk 1]
```diff
@@ -238,7 +238,7 @@ void shouldAddEmptyListValueToNewKeysWhenSomeKeysAreAlreadyPresent() {
             assertThat(map).extractingByKeys(ALL_PARTIES, PARENTS_AND_RESPONDENTS, COURT, OTHERS)
                 .allMatch(List::isEmpty);
             assertThat(map).extractingByKeys(LOCAL_AUTHORITY, CAFCASS)
-                .allMatch(list -> list.equals(emptyListOfElement()));
+                .allMatch(elemenList -> elemenList.equals(emptyListOfElement()));
```
Comment (lewisbirks) [inline service/src/test/java/uk/gov/hmcts/reform/fpl/service/CommonDirectionServiceTest.java:241]: Seems like an unnecessary change for this PR

File: service/build.gradle [hunk 2]
```diff
@@ -181,33 +188,40 @@ def versions = [
...
+  implementation group: 'org.springframework.security', name: 'spring-security-config'
+  implementation group: 'org.springframework.security', name: 'spring-security-oauth2-resource-server'
+  implementation group: 'org.springframework.security', name: 'spring-security-oauth2-jose'
+
+  implementation group: 'org.springframework.cloud', name: 'spring-cloud-starter-netflix-hystrix', version: '2.2.2.RELEASE'
+
+  implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: versions.log4j
+  implementation group: 'org.apache.logging.log4j', name: 'log4j-to-slf4j', version: versions.log4j
+
+  implementation group: 'com.nimbusds', name: 'nimbus-jose-jwt', version: '8.17' // override to fix vulnerability issue (CVE-2019-17195)
```
Comment (lewisbirks) [inline service/build.gradle:211]: Do we still need this or has spring security (I think that is the right package) been updated to no longer have this issue?

Comment (Gogetter) [reply service/build.gradle]: good spot. updated

File: service/build.gradle [hunk 3]
```diff
@@ -105,8 +109,8 @@ task functional(description: 'Runs the functional tests.') {
 checkstyle {
   maxWarnings = 0
   toolVersion = "8.25"
-  // need to set configDir to rootDir otherwise submodule will use submodule/config/checkstyle
-  configDir = file("$rootDir/config/checkstyle")
+  // need to set configDirectory to rootDir otherwise submodule will use submodule/config/checkstyle
+  configDirectory = file("$rootDir/config/checkstyle")
```
Comment (lewisbirks) [inline service/build.gradle:113]: intellij complaining about `Access to 'configDirectory' exceeds its access rights` and `Cannot assign 'File' to 'DirectoryProperty' `
Maybe
```suggestion
  configDirectory.set(file("$rootDir/config/checkstyle"))
```

File: service/build.gradle [hunk 4]
```diff
@@ -221,14 +234,16 @@ dependencies {
...
 
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-mail'
 
-  testCompile (group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: versions.springBoot) {
+  testImplementation group: 'org.apache.pdfbox', name: 'pdfbox', version: '2.0.19'
+
+  testCompile (group: 'org.springframework.boot', name: 'spring-boot-starter-test') {
     exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
     exclude group: 'junit', module: 'junit'
     exclude group: 'org.mockito', module: 'mockito-core'
```
Comment (lewisbirks) [inline service/build.gradle:249]: Random question but do you know why we are excluding mockito here?
Is there something wrong with the version that is included with `spring boot starter test` that means 3.0.0 is wanted?

File: service/build.gradle [hunk 5]
```diff
@@ -221,14 +234,16 @@ dependencies {
...
 
-  compile group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.1'
+  implementation group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.1'
 
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-mail'
 
-  testCompile (group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: versions.springBoot) {
+  testImplementation group: 'org.apache.pdfbox', name: 'pdfbox', version: '2.0.19'
+
+  testCompile (group: 'org.springframework.boot', name: 'spring-boot-starter-test') {
```
Comment (lewisbirks) [inline service/build.gradle:246]: Also is there a reason for the removal of `version: versions.springBoot`

Comment (Gogetter) [reply service/build.gradle:246]: removed this to take advantage of the spring dependency management

File: service/build.gradle [hunk 6]
```diff
@@ -161,14 +165,17 @@ sonarqube {
...
+  mavenLocal()
   jcenter()
+  mavenCentral()
+
   maven {
     url "https://dl.bintray.com/hmcts/hmcts-maven"
   }
+
   maven {
     url  "http://repo.spring.io/milestone"
```
Comment (lewisbirks) [inline service/build.gradle:177]: https?

File: service/build.gradle (hunk 3 above)
Comment (Gogetter) [reply service/build.gradle:113]: yeah, saw that wasn't sure if the pipeline would play along with `configDirectory.set` as `configDirectory = file("$rootDir/config/checkstyle")` was accepted. As the pipeline did not accept making `def inStrictOrder` static as suggested by IJ

Comment (lewisbirks) [reply service/build.gradle:113]: ahh fair enough

File: service/build.gradle (hunk 4 above)
Comment (Gogetter) [reply service/build.gradle]: hmmm, not sure but looking through the git history it was done as part of a gradle plugin update. Here is the commit hash `914d8e5cb5db6ce5ca474516f17b4fca2943a87e`

File: service/build.gradle [hunk 7]
```diff
@@ -244,14 +244,9 @@ dependencies {
   testImplementation group: 'org.apache.pdfbox', name: 'pdfbox', version: '2.0.19'
 
   testCompile (group: 'org.springframework.boot', name: 'spring-boot-starter-test') {
-    exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
```
Comment (lewisbirks) [inline service/build.gradle:232]: This was previously excluded due to issues with the mutation testing, might want to check that it still runs if you are removing this part

File: service/build.gradle [hunk 8]
```diff
@@ -1,14 +1,19 @@
...
+  id "java"
+  id "java-library"
   id 'application'
   id 'checkstyle'
   id 'jacoco'
   id 'io.spring.dependency-management' version '1.0.9.RELEASE'
   id 'org.springframework.boot' version '2.3.0.RELEASE'
   id 'com.github.ben-manes.versions' version '0.28.0'
-  id 'org.sonarqube' version '3.0'
+  id 'org.sonarqube' version '2.8'
```
Comment (Gogetter) [inline service/build.gradle:10]: did this due to a recent error 
```
Plugin Repositories (could not resolve plugin artifact 'org.sonarqube:org.sonarqube.gradle.plugin:3.0')
```

File: gradlew [hunk 9]
```diff
@@ -82,6 +82,7 @@ esac
 
 CLASSPATH=$APP_HOME/gradle/wrapper/gradle-wrapper.jar
 
+
```
Comment (toireasam) [inline gradlew:85]: this intentional?

Comment (Gogetter) [reply gradlew:85]: hmm, I think its because it was generated using `gradle wrapper --gradle-version 6.4.1`

Comment (toireasam) [reply gradlew:85]: np :)

**Pull request merged**